### DISCUSSION TITLE
Podman setting ownership of data and config directories to wrong uid & gid

### DISCUSSION THREAD

**Discussion by CybeatB** [abraunegg/onedrive#2173] [Q&A] [answered]
I followed the Podman setup instructions to the letter, except that I set `${HOME}/.config/onedrive` to be my config directory, instead of using a podman volume. My data directory is `${HOME}/OneDrive`.

```
export ONEDRIVE_UID=1000
export ONEDRIVE_GID=1000
podman run -it --name onedrive --user "${ONEDRIVE_UID}:${ONEDRIVE_GID}" \
    -v "${HOME}/.config/onedrive:/onedrive/conf:U,Z" \
    -v "${HOME}/OneDrive:/onedrive/data:U,Z" \
    -e PODMAN=1 \
    --label "io.containers.autoupdate=image" \
    driveone/onedrive:latest
```

My UID and GID on the host system are 1000:1000, so I'm passing those values in via the `--user` flag.
However, as soon as the container starts running, my config and data directories are suddenly owned by seemingly-random uid/gid pairs, like 100999:100999 or 999:ping instead of 1000:1000.

I assume this means that podman is setting the uid and gid inside the container incorrectly? How do I fix it?

**Comment by abraunegg** [comment, marked as answer]
@CybeatB 
You need to potentially use the `--userns=keep-id` option within podman to keep the configured user|group id.

![image](https://user-images.githubusercontent.com/4956234/194464553-b2cb0656-c105-45f7-87bd-60146f47e486.png)

Reference:
* https://docs.podman.io/en/latest/markdown/podman-run.1.html

**Reply by CybeatB** [reply]
That seems to have worked, thank you very much!
